cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -162,6 +167,31 @@ public BodyConsumer deploy(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Creates an application with the specified name and app-id from an artifact.
+   */
+  @POST
+  @Path("/apps/{app-id}/versions/{version-id}/create")
+  public BodyConsumer createAppVersion(HttpRequest request, HttpResponder responder,
+                                         @PathParam("namespace-id") final String namespaceId,
+                                         @PathParam("app-id") final String appId,
+                                         @PathParam("version-id") final String versionId)
+    throws Exception {
+
+    ApplicationId applicationId = validateApplicationVersionId(namespaceId, appId, versionId);
+
+    if (!applicationLifecycleService.updateAppAllowed(applicationId))  {
+      responder.sendString(HttpResponseStatus.CONFLICT, "Cannot update the application because of the " +
+        "existing version(s)");
```

Comment:
Will this be confusing for the user since they can update the application sometimes when the versionId ends with "-SNAPSHOT"?